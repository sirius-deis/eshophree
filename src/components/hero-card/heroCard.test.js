import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import HeroCard from './heroCard';

describe('Card component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <HeroCard btnTitle='Card' />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render element with correct button title', () => {
    render(
      <MemoryRouter>
        <HeroCard btnTitle='Card' />
      </MemoryRouter>,
    );
    const title = screen.queryByText(/Card/);
    expect(title).toBeInTheDocument();
  });
  it('should render a link inside the button', () => {
    const { container } = render(<MemoryRouter><HeroCard btnTitle="Shop Now" /></MemoryRouter>);
    const linkElement = container.querySelector('a');
    expect(linkElement).toBeInTheDocument();
  });
});
