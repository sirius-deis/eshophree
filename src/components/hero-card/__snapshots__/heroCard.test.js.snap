// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
<div>
  <div
    class="sc-bgqQcB bVJoSM"
  >
    <img
      alt="card"
      src="https://source.unsplash.com/random/400×200?electronics"
    />
    <div
      class="sc-gTRrQi iOjQnv"
    >
      <button
        class="sc-ewnqHT sc-iVCKna hbuNTc hhqmXX"
        color="bg-color"
        type="button"
      >
        <a
          href="/"
        >
          Card
           →
        </a>
      </button>
    </div>
  </div>
</div>
`;
